OpenRA.Mods.Common/Pathfinder/PathSearch.cs

```diff
@@ -162,10 +162,18 @@ static CellInfoLayerPool LayerPoolForWorld(World world)
 			openQueue = new PriorityQueue<GraphConnection>(GraphConnection.ConnectionCostComparer);
 		}
 
-		void AddInitialCell(CPos location)
+		void AddInitialCell(CPos location, Func<CPos, int> customCost)
```

Comment:
Minor nit: Can we `Func<CPos, int> customCost = null` here and drop the explicit `null` from line 90? (Several of the other pathfinder related methods also default to `null` for this parameter.)

Reply:
For public surface where you want the parameter to be optional, I would agree. For private surface like this though I prefer to keep it explicit. It forces the caller to remember to pass the parameter along or make a conscious decision it is not needed.

Reply:
Ok, you know this code better than me (and also work with it more ^^).